File: clients/typescript/src/my-strategy.ts
import { PlayerView } from "./model/player-view";
import { Order } from "./model/order";
import { VehicleOrder } from "./model/vehicle-order";
import { Constants } from "./model/constants";
import { DebugInterface } from "./debug-interface";

export class MyStrategy {
    constructor(constants: Constants) {}
    async getOrder(playerView: PlayerView, debug_interface: DebugInterface | null): Promise<Order> {
        return new Order([]);
    }
    async debugUpdate(displayed_tick: number, debug_interface: DebugInterface) {}
    finish() {}
}

File: clients/haskell/src/MyStrategy.hs
module MyStrategy (MyStrategy, createMyStrategy, getOrder) where

import Control.Monad.Trans.State.Lazy (State)
import Model.Constants
import Model.PlayerView
import Model.Order
import Model.VehicleOrder

data MyStrategy = MyStrategy

createMyStrategy :: Constants -> MyStrategy
createMyStrategy _ = MyStrategy

getOrder :: PlayerView -> State MyStrategy Order
getOrder _ = return Order { vehicles = [] }